DEVDOCS-6170 - [New Doc] Guide for PCI on Checkout

bc-terra opened the pull request:
Providing a developer guide for updating custom checkout to include PCI DSS 4.0 Section 6.4.3 compliance features

<!-- Ticket number or summary of work -->
# [DEVDOCS-6170]


## What changed?
* PCI DSS 4.0 Section 6.4.3 requires the inclusion of nonce-based authentication and SRI for checkout pages
* BigCommerce provides this functionality via updated features in control panel settings and Open Checkouts
* This provides a guide for developers for the use of the functionality.

## Release notes draft
* Created new guide for setting up PCI compliant features on checkout

## Anything else?
SME review waived by GTM team.
Navigation update to be requested and managed separately.

ping {names}


[DEVDOCS-6170]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-6170?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

bc-0dp commented on docs/storefront/cart-checkout/open-checkouts/pci-compliance-guide.mdx:
```diff
@@ -0,0 +1,75 @@
+# Configuring Open Checkout to comply with PCI DSS 4.0 Section 6.4.3
+If you’re using a custom checkout based on a fork of the [checkout-js](https://github.com/bigcommerce/checkout-js) repository, you will need to make changes in order  to meet PCI DSS 4.0 compliance requirements. This guide outlines the steps and actions you need to perform.
```
order  to

There is double space character

bc-terra replied:
Removed extra space. Good spot on that.

bc-0dp commented on docs/storefront/cart-checkout/open-checkouts/pci-compliance-guide.mdx:
````diff
@@ -0,0 +1,75 @@
+</script>
+```
+<Callout type="info">
+ If you are using a multi-storefront setup and have a channel specific custom checkout, navigate to <em>Channel Manager > Storefronts > Edit settings > Checkout</em> instead.
````
Channel Manager is called Channels in the new UI

bc-terra replied:
Thanks for catching that. Updated to match current UI

bc-0dp reviewed (commented):
I have no real comments, and I think it looks great. One thing I notice though is the url is `open-checkouts` plural but the navigation in the dev docs is singular `open-checkout`

Don't think it's that important but it cought my eye

bc-terra commented on docs/storefront/cart-checkout/open-checkouts/pci-compliance-guide.mdx:
```diff
@@ -0,0 +1,75 @@
+## Useful References
+- [Preparing Your Store for PCI DSS 4.0](https://www.bigcommerce.com.au/blog/preparing-your-store-for-pci-dss-4-0/)
+- [Security and Privacy Settings](https://support.bigcommerce.com/s/article/Security-and-Privacy-Settings)
+- [Subresource Integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity)
```
Confirmed not dead. May be blocked by non-human traffic.

bc-terra commented on docs/storefront/cart-checkout/open-checkouts/pci-compliance-guide.mdx:
```diff
@@ -0,0 +1,75 @@
+- [Preparing Your Store for PCI DSS 4.0](https://www.bigcommerce.com.au/blog/preparing-your-store-for-pci-dss-4-0/)
+- [Security and Privacy Settings](https://support.bigcommerce.com/s/article/Security-and-Privacy-Settings)
+- [Subresource Integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity)
+- [nonce](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/nonce)
```
Confirmed not dead. May be blocked to non-human traffic

bc-terra commented on docs/storefront/cart-checkout/open-checkouts/pci-compliance-guide.mdx:
```diff
@@ -0,0 +1,75 @@
+- [Security and Privacy Settings](https://support.bigcommerce.com/s/article/Security-and-Privacy-Settings)
+- [Subresource Integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity)
+- [nonce](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/nonce)
+- [Unsafe-eval Expressions](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src#unsafe_eval_expressions)
```
Confirmed not dead. May be blocked to non-human traffic

bc-terra commented:
> * In Step 2, it assumes there is prior knowledge about how the `openssl` command works. Consider adding a brief explanation of what `openssl` is doing.
> * In Step 4, you might want to add a clarification that nonce values are dynamically generated for each request and should not be hardcoded. Also explain why nonce-based authorization is required under PCI DSS 4.0.

Added reference for OpenSSL command line to the links, as this doc does assume some level of knowledge, but resources are always helpful.

Added statement about purpose of nonce-based authorization with respect to PCI 4.0 and a callout that values are randomly generated and not to be stored.

bc-traciporter reviewed (approved):
Nice work @bc-terra
